src/hyperion/experiment_plans/rotation_scan_plan.py

```diff
@@ -264,6 +283,13 @@ def rotation_scan(
     def rotation_scan_plan_with_stage_and_cleanup(
         params: RotationScan,
     ):
+        assert composite.aperture_scatterguard.aperture_positions is not None
+        yield from move_aperture(
```

Comment:
Must: I think this should probably go in or next to `setup_sample_environment`. We should be staging the detector as the first thing we do then asynchronously moving this into the correct place, we then only need to wait for it to have finished moving just before we do `rotation_scan_plan`

Reply:
Yeah good point!